### PR TITLE
Loader Architecture changes

### PR DESCRIPTION
**Pull request by pavitramohan**
Loader Architecture changes

### CODE REVIEW

**Comment by srinivastuta** [comment]
Pavitra, ConfigurationService should be backward compatible. It should execute the existing code to find the resource file under resources and if it cannot find, then it should go through additional logic of loading it from new loading architecture

File: jaffa-soa/source/java/org/jaffa/loader/TransactionManager.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+     * @param context with which repository to be associated with
+     */
+    public void unregisterTypeInfo(String typeInfoName, String context) {
+        typeInfoRepository.unregister(typeInfoName, context);
+    }
+
+    /**
+     * unregisters all the transactions and typeInfo in the xml from the repository
+     * @param uri for the xml location
+     */
```
Comment (srinivastuta) [inline jaffa-soa/source/java/org/jaffa/loader/TransactionManager.java]: @todo

File: jaffa-core/source/java/org/jaffa/loader/XmlLoader.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    /**
+     * gets the context associated with the path
+     * @param path where the resource is located
+     * @return context associated with the path
+     */
+    public String getContext(String path){
+        String context = "default";
+
+        if(defaultContexts != null) {
```
Comment (pavitramohan) [inline jaffa-core/source/java/org/jaffa/loader/XmlLoader.java]: I think if it is autowired with null value and I am just checking to see it should not throw any errors and handle accordingly..

Comment (srinivastuta) [inline jaffa-core/source/java/org/jaffa/loader/XmlLoader.java]: You are initializing it at line 69, regardless of autowired

File: jaffa-soa/src/test/org/jaffa/transaction/services/AllXmlLoadTest.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package org.jaffa.transaction.services;
```
Comment (srinivastuta) [inline jaffa-soa/src/test/org/jaffa/transaction/services/AllXmlLoadTest.java]: The folder structure is incorrect. This should be under src/test/java/org/jaffa/transaction/services

**Pull request merged**